src/internal/m365/collection/drive/collections_tree_test.go

```diff
@@ -0,0 +1,521 @@
+		mbh,
+		tenant,
+		idname.NewProvider(user, user),
+		func(*support.ControllerOperationStatus) {},
+		control.Options{ToggleFeatures: control.Toggles{
+			UseDeltaTree: true,
+		}},
+		count.New())
+}
+
+func fullOrPrevPath(
+	t *testing.T,
+	coll data.BackupCollection,
+) path.Path {
+	var collPath path.Path
+
+	if coll.State() != data.DeletedState {
+		collPath = coll.FullPath()
+	} else {
+		collPath = coll.PreviousPath()
+	}
+
+	require.False(
+		t,
+		len(collPath.Elements()) < 4,
+		"malformed or missing collection path")
+
+	return collPath
+}
+
+func pagerForDrives(drives ...models.Driveable) *apiMock.Pager[models.Driveable] {
+	return &apiMock.Pager[models.Driveable]{
+		ToReturn: []apiMock.PagerResult[models.Driveable]{
+			{Values: drives},
+		},
+	}
+}
+
+func makePrevMetadataColls(
+	t *testing.T,
+	mbh BackupHandler,
+	previousPaths map[string]map[string]string,
+) []data.RestoreCollection {
+	pathPrefix, err := mbh.MetadataPathPrefix(tenant)
+	require.NoError(t, err, clues.ToCore(err))
+
+	prevDeltas := map[string]string{}
+
+	for driveID := range previousPaths {
+		prevDeltas[driveID] = idx(delta, "prev")
+	}
+
+	mdColl, err := graph.MakeMetadataCollection(
+		pathPrefix,
+		[]graph.MetadataCollectionEntry{
+			graph.NewMetadataEntry(bupMD.DeltaURLsFileName, prevDeltas),
+			graph.NewMetadataEntry(bupMD.PreviousPathFileName, previousPaths),
+		},
+		func(*support.ControllerOperationStatus) {},
+		count.New())
+	require.NoError(t, err, "creating metadata collection", clues.ToCore(err))
+
+	return []data.RestoreCollection{
+		dataMock.NewUnversionedRestoreCollection(t, data.NoFetchRestoreCollection{Collection: mdColl}),
+	}
+}
+
+func compareMetadata(
+	t *testing.T,
+	mdColl data.Collection,
+	expectDeltas map[string]string,
+	expectPrevPaths map[string]map[string]string,
+) {
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	colls := []data.RestoreCollection{
+		dataMock.NewUnversionedRestoreCollection(t, data.NoFetchRestoreCollection{Collection: mdColl}),
+	}
+
+	deltas, prevs, _, err := deserializeAndValidateMetadata(
+		ctx,
+		colls,
+		count.New(),
+		fault.New(true))
+	require.NoError(t, err, "deserializing metadata", clues.ToCore(err))
+	assert.Equal(t, expectDeltas, deltas, "delta urls")
+	assert.Equal(t, expectPrevPaths, prevs, "previous paths")
+}
+
+// for comparisons done by collection state
+type stateAssertion struct {
+	itemIDs []string
+	// should never get set by the user.
+	// this flag gets flipped when calling assertions.compare.
+	// any unseen collection will error on requireNoUnseenCollections
+	sawCollection bool
+}
+
+// for comparisons done by a given collection path
+type collectionAssertion struct {
+	doNotMerge    assert.BoolAssertionFunc
+	states        map[data.CollectionState]*stateAssertion
+	excludedItems map[string]struct{}
+}
+
+type statesToItemIDs map[data.CollectionState][]string
+
+func newCollAssertion(
+	doNotMerge bool,
+	itemsByState statesToItemIDs,
+	excludeItems ...string,
```

Comment:
I don't think this is the correct place to be taking excludes. Excludes are global to the entire test whereas this seems tied to the particular collection. I think this will end up clobbering previous excludes if this helper is used for multiple collections

Reply:
hmm, fair.  I think I'll follow up on this later, since it's unused for the next handful of PRs.  When I get into rebuilding all of the mocked e2e tests (ie: the Get() tests) it'll be a natural time to revisit the design.